Sources/Core/AWSClientRuntime/Auth/AuthSchemes/SigV4AuthScheme.swift

```diff
@@ -16,34 +16,53 @@ public struct SigV4AuthScheme: ClientRuntime.AuthScheme {
 
     public func customizeSigningProperties(signingProperties: Attributes, context: HttpContext) -> Attributes {
         var updatedSigningProperties = signingProperties
-        updatedSigningProperties.set(key: AttributeKeys.bidirectionalStreaming, value: context.isBidirectionalStreamingEnabled())
+
+        updatedSigningProperties.set(key: AttributeKeys.signingAlgorithm, value: .sigv4)
+
+        updatedSigningProperties.set(
+            key: AttributeKeys.bidirectionalStreaming,
+            value: context.isBidirectionalStreamingEnabled()
+        )
         updatedSigningProperties.set(key: AttributeKeys.signingName, value: context.getSigningName())
         updatedSigningProperties.set(key: AttributeKeys.signingRegion, value: context.getSigningRegion())
-        updatedSigningProperties.set(key: AttributeKeys.signingAlgorithm, value: .sigv4)
 
-        // Expiration is only used for presigning URLs. E.g., in AWSS3, and in AWSPolly.
-        updatedSigningProperties.set(key: AttributeKeys.expiration, value: 0)
-        // AWSSignatureType.requestQueryParams is only used for S3 GetObject and PutObject
-        // Out of all AWSSignatureType cases, only two are used. .requestHeaders and .requestQueryParams.
-        // .requestHeaders is the deafult signing used for all AWS operations except S3 customizations.
-        updatedSigningProperties.set(key: AttributeKeys.signatureType, value: .requestHeaders)
+        // Expiration is only used for presigning (presign request flow or presign URL flow).
+        updatedSigningProperties.set(key: AttributeKeys.expiration, value: context.getExpiration())
 
-        // SigningFlags
+        // AWSSignatureType.requestQueryParams is only used for presign URL flow.
+        // Out of the AWSSignatureType enum cases, only two are used. .requestHeaders and .requestQueryParams.
+        // .requestHeaders is the deafult signing used for AWS operations.
         let serviceName = context.getServiceName()
+        let isS3PresignURLOp = serviceName == "S3" && ["getObject", "putObject"].contains(context.getOperation())
```

Comment:
Since this same line is reused we should have AuthScheme Helper function file or AuthScheme constants so that we minimize duplicated code

Reply:
Factored out literals into a helper struct to reflect suggestion.